Sources/Workspace/Workspace.swift

```diff
@@ -335,8 +335,15 @@ public class Workspace {
     }
 
     /// Resets the entire workspace by removing the data directory.
-    public func reset() throws {
-        try managedDependencies.reset()
+    ///
+    /// - Parameters:
+    ///     - diagnostics: The diagnostics engine that reports errors, warnings
+    ///       and notes.
+    public func reset(with diagnostics: DiagnosticsEngine) {
+        guard diagnostics.wrap({ try managedDependencies.reset() }) else {
```

Comment:
Yeah, thats because `managedDependencies.reset()` ends up calling `FileSystem.writeFileContents` which throws, but `repositoryManager.reset()` calls `FileSystem.removeFileTree` which doesn't throw. Don't know what you want to do about that.

Sounds weird because `FileManager.removeItem(atPath:)` could potentially throw (I think) if a file under that path is opened by another process. No? perhaps `FileSystem.removeFileTree` should propagate that error.